src/main/java/net/minecraftforge/registries/ObjectHolder.java

```diff
@@ -16,7 +16,9 @@
 /**
  * ObjectHolder can be used to automatically populate public static final fields with entries
  * from the registry. These values can then be referred within mod code directly.
+ * @deprecated Use {@link DeferredRegister} instead.
  */
+@Deprecated(since = "1.21.4", forRemoval = true)
```

Comment:
So for most cases people just want DeferredRegister. As they want suppliers for their own objects.
Tho more specifically if you don't want to register an object and just want a holder for the 'current dirt' then you want RegistryObject.
I think it would be better to write a sample of both of these cases in the comment when deprecating so people know where to go for their usecase.